### ISSUE TITLE
_onReceivedInitSessionResult crash

### ISSUE THREAD

**Issue by lprhodes** [BranchMetrics/React-Native-Deep-Linking-SDK#12]
Hi,

I'm getting a crash inside _onReceivedInitSessionResult on iOS.

When trying to perform forEach on this._patientInitSessionObservers it's finding _patientInitSessionObservers to be undefined. 

I believe this is because of how you're using 'this' inside arrow functions and some kind of conflict with babel.

Using this inside arrow functions keeps the parent scope for this and using them as methods inside classes means 'this' is pointing to the DedicatedWorkerGlobalScope

Changing it to be a normal method fixes the issue and this then points to the Branch object as you'd expect.

e.g.

```_onReceivedInitSessionResult = (result) => {```

becomes

```_onReceivedInitSessionResult (result) {```

Which has the benefit of being simpler too.

**Comment by lprhodes** [comment]
The nativeEventEmitter also needs to be bound with 'this'

i.e.

```nativeEventEmitter.addListener('RNBranch.initSessionFinished', this._onReceivedInitSessionResult.bind(this));``

**Comment by rt2zz** [comment]
@lprhodes that is strange. What version of react-native/babel are you on? The arrow functions should always have the lexical scope of the class they were defined on.

If you can I would recommend switching over to https://github.com/BranchMetrics/react-native-branch where ongoing development is happening and a lot of improvements have landed. We could use the extra eyes on that repo and if the problem persists there we can dig deeper.

**Comment by lprhodes** [comment]
The latest for both.

I've just installed react-native-branch and get the same issue there - I've upgraded all nam modules to their latest.

I'm using babel-preset-react-native-stage-0 which is the only thing I can think of that would different form a standard install.

https://dev.branch.io/getting-started/sdk-integration-guide/guide/react/ points to this version. It'd be great if that could be updated if this version is ready and a note on this git repo too.
